services/src/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/flows/FlowCrudSpec.groovy

```diff
@@ -426,6 +427,79 @@ class FlowCrudSpec extends BaseSpecification {
         }
     }
 
+    def "Able to validate flow with protected path"() {
+        given: "Two active not neighboring switches with two possible paths at least"
+        def switches = topology.getActiveSwitches()
+        def allLinks = northbound.getAllLinks()
+        List<List<PathNode>> possibleFlowPaths = []
+        def (Switch srcSwitch, Switch dstSwitch) = [switches, switches].combinations()
+                .findAll { src, dst -> src != dst }.find { Switch src, Switch dst ->
+            possibleFlowPaths = database.getPaths(src.dpId, dst.dpId)*.path.sort { it.size() }
+            List<List<Switch>> pathSwitches = possibleFlowPaths.collect { pathHelper.getInvolvedSwitches(it) }
+            Integer numberOfDifferentSwitchInPath = 0
+            for (int i = 0; i < pathSwitches.size() - 1; i++) {
+                if (!pathSwitches[i][1..-2].containsAll(pathSwitches[i + 1][1..-2])) {
+                    numberOfDifferentSwitchInPath += 1
+                }
+            }
+            allLinks.every { link ->
+                !(link.source.switchId == src.dpId && link.destination.switchId == dst.dpId)
+            } && possibleFlowPaths.size() >= 2 && numberOfDifferentSwitchInPath >= 2
+        } ?: assumeTrue("No suiting switches found", false)
+
+        and: "Create a flow with protected path"
+        def flow = flowHelper.randomFlow(srcSwitch, dstSwitch)
+        flow.allocateProtectedPath = true
+
+        when: "Create a flow with protected path"
+        flowHelper.addFlow(flow)
+
+        then: "Flow with protected path is created"
+        def flowPathInfo = northbound.getFlowPath(flow.id)
+        flowPathInfo.protectedPath
+
+        and: "Validation of flow with protected path must be successful"
+        northbound.validateFlow(flow.id).each { assert it.discrepancies.empty }
+
+        and: "No rule discrepancies on every switch of the flow on the main path"
+        def mainPath = PathHelper.convert(flowPathInfo)
+        def mainSwitches = pathHelper.getInvolvedSwitches(mainPath)
+        if (flow.source.datapath == flow.destination.datapath) {
+            mainSwitches << topology.activeSwitches.find { it.dpId == flow.source.datapath }
+        }
+        mainSwitches.each { verifySwitchRules(it.dpId) }
+
+        and: "No rule discrepancies on every switch of the flow on the protected path)"
+        def protectedPath = PathHelper.convert(flowPathInfo.protectedPath)
+        def protectedSwitches = pathHelper.getInvolvedSwitches(protectedPath)
+        protectedSwitches.each { verifySwitchRules(it.dpId) }
+
+        when: "Swap path on the flow"
```

Comment:
You have a separate test doing the same steps, don't you? Or is it different?

Reply:
I am with you on this one
I will remove this test

Reply:
I will extend the 'Able to swap flow path' test in the ProtectedPath spec